### ISSUE TITLE
Performance : countRecords should take the where clause

### ISSUE THREAD

**Issue by mickaelgudin** [Sarveshgithub/sfdc-lwc-lightning-datatable#7]
The method countRecords does not take the where clause into account if set.

The solution should be simple and it'd enhance the performance : 
- add whereClause as param in apex class RelatedList (method countRecords)
- use appendWhere from lwc component lwcRelatedList to send the new parameter whereClause when calling countRecords

I'll implement it in the coming weeks.

**A possible new feature when this solution is implemented :**
- Append the total number next to the title(but optional)
![image](https://user-images.githubusercontent.com/39730173/157103859-165246eb-7f5a-4c30-85bf-ac999f0ab1b0.png)

**Comment by Sarveshgithub** [comment]
@mickaelgudin  Yes Looks nice feature to have a count of records beside of tiile.
I checked the code. in connectedcallback(), there is two apex class at same time getRecords() and countRecords(). We can avoid this two calls and make into single call. Anyway, we need count records for pagination. to append count in the title, just hide/show count value based on  Design Attribute.
Example: 
```
'count' parameter passed from LWC, initially it is zero
getRecords( Integer count) {
  If(count == 0) {
   countRecords()
  }
// In Wrapper class , need to add new key for count.
} 
```

Let me know if any questions.

**Comment by mickaelgudin** [comment]
@Sarveshgithub The same idea crossed my mind.

The problem I see is if the component is used on a large number of record. 
The two calls enable us to reach 50 000 rows(1 transaction / call), whereas a single call would limit the maximum to 25 000 rows.

Of course this is the worst case scenario, but since the current behavior allow this I think it's best to keep the two calls.

A future feature could enhance the pagination, it would not call fetchRecords and it would be based on the array index of all records. Then we could make a single initial call (in connectedcallback), totalCount would then be the length of the records array.

But since it would require thorough tests I want to stick to the first solution.
Since my main objective is to implement #8, because I had to implement the count in the title when I used this component on a client project.
